extensions/wikia/PowerUser/maintenance/PowerUserRemoveAdminsMaintenance.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Maintenance script for removing poweruser_admin
+ * property if a user lost his sysop rights.
+ *
+ * @package MediaWiki
+ * @addtopackage maintenance
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ *
+ *
+ */
+
+require_once( __DIR__.'/../../../../maintenance/Maintenance.php' );
+
+use Wikia\PowerUser\PowerUser;
+use Wikia\Logger\WikiaLogger;
+
+class PowerUserRemoveAdminsMaintenance extends Maintenance {
+	/**
+	 * Do the actual work. All child classes will need to implement this
+	 */
+	public function execute() {
+		$aAdminsIds = $this->getPowerUserAdminsIds();
+		$aFormerAdminsIds = $this->getFormerAdmins( $aAdminsIds );
+
+		foreach ( $aFormerAdminsIds as $iFormerAdminId ) {
+			$oPowerUser = new PowerUser( User::newFromId( $iFormerAdminId ) );
+			if ( $oPowerUser->removePowerUserProperty( PowerUser::TYPE_ADMIN ) ) {
+				WikiaLogger::instance()->info( PowerUser::LOG_MESSAGE, [
+					'type' => PowerUser::TYPE_ADMIN,
+					'action' => 'remove',
+				]);
+			}
+		}
+	}
+
+	private function getPowerUserAdminsIds() {
+		global $wgExternalSharedDB;
+
+		$oDB = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+
+		$aAdminsIds = ( new WikiaSQL() )
+			->SELECT( 'up_user' )
+			->FROM( 'user_properties' )
+			->WHERE( 'up_property' )->EQUAL_TO( PowerUser::TYPE_ADMIN )
+			->AND_( 'up_value' )->EQUAL_TO( '1' )
+			->runLoop( $oDB, function( &$aAdminsIds, $oRow ) {
+				$aAdminsIds[] = intval( $oRow->up_user );
+			});
+
+		return $aAdminsIds;
+	}
+
+	private function getFormerAdmins( $aAdminsIds ) {
+		global $wgSpecialsDB;
+		$oDB = wfGetDB( DB_SLAVE, [], $wgSpecialsDB );
+
+		$aFormerAdminsIds = [];
+		foreach ( $aAdminsIds as $iAdminId ) {
+			$iFormerAdminId = ( new WikiaSQL() )
+				->SELECT( 'user_id', 'all_groups' )
+				->FROM( 'events_local_users' )
+				->WHERE( 'user_id' )->EQUAL_TO( $iAdminId )
+				->LIMIT( 1 )
```

Comment:
Should this limit be here? Only checking one wikia won't do the trick, but some users will have entries for every wikia (like staff members), so would have 400k+ rows.